go.mod

```diff
@@ -3,54 +3,61 @@ module github.com/hashicorp/nomad-autoscaler
 go 1.19
 
 require (
-	github.com/Azure/azure-sdk-for-go v64.1.0+incompatible
-	github.com/Azure/go-autorest/autorest v0.11.27
-	github.com/Azure/go-autorest/autorest/azure/auth v0.5.11
+	github.com/Azure/azure-sdk-for-go v68.0.0+incompatible
+	github.com/Azure/go-autorest/autorest v0.11.29
+	github.com/Azure/go-autorest/autorest/azure/auth v0.5.12
 	github.com/Azure/go-autorest/autorest/date v0.3.0
-	github.com/DataDog/datadog-api-client-go v1.14.0
-	github.com/armon/go-metrics v0.3.11
-	github.com/aws/aws-sdk-go-v2 v1.16.4
-	github.com/aws/aws-sdk-go-v2/config v1.15.8
-	github.com/aws/aws-sdk-go-v2/service/autoscaling v1.23.2
-	github.com/golang/protobuf v1.5.2
-	github.com/google/go-cmp v0.5.8
-	github.com/hashicorp/consul/api v1.8.0
-	github.com/hashicorp/go-hclog v0.16.0
+	github.com/DataDog/datadog-api-client-go v1.16.0
+	github.com/armon/go-metrics v0.4.1
+	github.com/aws/aws-sdk-go-v2 v1.18.0
+	github.com/aws/aws-sdk-go-v2/config v1.18.22
+	github.com/aws/aws-sdk-go-v2/service/autoscaling v1.28.6
+	github.com/golang/protobuf v1.5.3
+	github.com/google/go-cmp v0.5.9
+	github.com/hashicorp/consul/api v1.20.0
+	github.com/hashicorp/go-hclog v1.5.0
 	github.com/hashicorp/go-msgpack v1.1.5
 	github.com/hashicorp/go-multierror v1.1.1
-	github.com/hashicorp/go-plugin v1.0.1
-	github.com/hashicorp/hcl/v2 v2.10.0
-	github.com/hashicorp/nomad/api v0.0.0-20220519231241-2b054e38e91a
+	github.com/hashicorp/go-plugin v1.4.9
+	github.com/hashicorp/hcl/v2 v2.16.2
+	github.com/hashicorp/nomad/api v0.0.0-20230502200911-4d9b969d816b
 	github.com/mitchellh/cli v1.1.2
-	github.com/mitchellh/copystructure v1.1.2
+	github.com/mitchellh/copystructure v1.2.0
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/prometheus/client_golang v1.12.2
-	github.com/prometheus/common v0.32.1
-	github.com/stretchr/testify v1.7.1
-	google.golang.org/api v0.80.0
-	google.golang.org/grpc v1.46.0
-	google.golang.org/protobuf v1.28.0
+	github.com/prometheus/client_golang v1.15.1
+	github.com/prometheus/common v0.42.0
+	github.com/stretchr/testify v1.8.2
+	google.golang.org/api v0.120.0
+	google.golang.org/grpc v1.54.0
+	google.golang.org/protobuf v1.30.0
```

Comment:
Did all of these updates come from Nomad? While they look OK we've had some backwards incompatibility issues on dependency upgrade in the past, so it may be better to upgrade them in separate commits so we can bisect problems.

It would be also good to run a batch of [cluster scaling](https://developer.hashicorp.com/nomad/tutorials/autoscaler/horizontal-cluster-scaling) tests after the version bump.

Reply:
Successfully run, without anything to report.